src/Interpreter.hs

```diff
@@ -55,13 +56,17 @@ withInterpreter flags action = do
   let
     args = flags ++ [
         "--interactive"
+      , xTemplateHaskell
 #if __GLASGOW_HASKELL__ >= 802
```

Comment:
What is the meaning of such an `#if`?  The `#if` is evaluated when `doctest` is compiled, and we use such `#if`s all over the place to ensure compilation with different GHCs.  But this `#if` does not make to seem a difference on the _compilation_ of `doctest`, but when I _run_ `doctest`.
Could it be a problem when I run `doctest` with a different GHC version than it was compiled with?

Reply:
`doctest` uses both `ghc` (the library) and `ghc --interactive` (the executable).  The former is used to extract Haddock comments, the later is used to check examples.  If we would only depend on the executable, we could just use whatever `ghc` is on the `PATH`.  But with `ghc` the library we can only depend on one version (specifically, the version we compile `doctest` with).  For that reason we fix both, `ghc` the library and `ghc` the executable to the `ghc` version we compile against at compile time.  The library version is fixed simply by depending on it. To fix the executable version we use `ghc-paths`.

Reply:
Ah, this makes sense if `doctest` sits within a per-GHC buildtree.  
(For some reason, I have `doctest` in my PATH, but that does not make sense in such a scenario.)